contracts/fixed_price_rev1.1.scilla

```diff
@@ -688,7 +688,8 @@ transition FulfillOrder(
   RequireNotPaused;
   RequireAllowedUser _sender;
   RequireAllowedUser dest;
```

Comment:
Should we also remove this check?

Reply:
Hi @anton-trunov why do we need to remove the `RequireAllowedUser dest` check?

Reply:
I believe there should be exactly one check for the validity of a destination. So if we need to use allowlists for this, then `RequireAllowedUser` should be called from `RequireValidDestination`, otherwise it's error-prone w.r.t. future modifications of the contract: someone might forget to use two checks for destination validity as opposed to just one. One more consideration: if the buyer already got an approval from the contract, then they should be able to set any destination of their choice, so no need to use allowlists for that.

Reply:
@anton-trunov we can keep `RequireAllowedUser` as a separate call, `RequireValidDestination` is called in other places (see `SetServiceFeeRecipient`) which does need the address to be in the allowed list.

> if the buyer already got an approval from the contract, then they should be able to set any destination of their choice, so no need to use allowlists for that

The biz requirement is that the destination also need to be in the allowed list (not just the buyer).